ggshield/core/oauth.py

```diff
@@ -358,11 +364,18 @@ def _print_login_success(self) -> None:
         else:
             str_date = "never"
 
+        expiration_warning = ""
+        if self._expire_at_downsized:
+            expiration_warning = (
+                " Warning: the expiration date has been enforced due to your workspace's"
+                " setting for the maximum lifetime of personal access tokens."
+            )
+
         message = (
             "Success! You are now authenticated.\n"
             "The personal access token has been created and stored in your ggshield config.\n\n"
             f"token name: {self._token_name}\n"
-            f"token expiration date: {str_date}\n\n"
+            f"token expiration date: {str_date}{expiration_warning}\n\n"
```

Comment:
I am afraid the output for this won't be very readable. I suggest removing it from there and using `display_warning()` from `ggshield.core.text_utils` to print the warning message after the call to `click.echo(message)`.

Something like this:
```python
click.echo(message)
if self.expire_at_downsized:
    display_warning(
        "Warning: the expiration date has been enforced due to your workspace's"
        " setting for the maximum lifetime of personal access tokens."
    )
```

Reply:
Sounds better. @elacaille18 is it good for you?

Reply:
I am not sure to agree

the final sentence of the message is about something else... :
```
You do not need to run "ggshield auth login" again. Future requests will automatically use the token.
```

I want the warning to be closed to the expiration, it is better UX. So I suggest to put it just below

```
message = (
            "Success! You are now authenticated.\n"
            "The personal access token has been created and stored in your ggshield config.\n\n"
            f"token name: {self._token_name}\n"
            f"token expiration date: {str_date}\n\n"
            {expiration_warning}\n\n
            'You do not need to run "ggshield auth login" again. Future requests will automatically use the token.'
        )
```

Reply:
Sounds good.